### PR TITLE
Pick and place triggers

### LINKED ISSUE #1064: Allow trigger for pick&place
When correctly placed: trigger next step. If there are several pick&place augmentations: allow auto progression only for when the last one is correctly placed.

### PR DESCRIPTION
**Pull request by robhillman97**
Adds triggers to pick and place

closes #1064 & #1041

### CODE REVIEW

**Review by cawke — CHANGES_REQUESTED** [review]
Follow the code style.

**Review by cawke — COMMENTED** [review]
Please try running the tests locally.

**Comment by fwild** [comment]
tests are now running through - though the previous test failures were in traced to "Tests.SetStartingPositionTests+<OnPlayerReset_RotationUpdated>d__4.MoveNext () [0x0013a] in /github/workspace/Assets/UnitTests/PlayModeTests/SetStartingPositionTests.cs" in the lines 101/127/154/181, so is either a glitch, or has been fixed elsewhere, but seems to have nothing to do with the code changes in this push.

File: Assets/MirageXR/Tests/NewUI/ContentListView_v2.cs [hunk 1]
```diff
@@ -192,6 +192,7 @@ private void UpdateView()
             {
                 var obj = Instantiate(_contentListItemPrefab, _listContent);
                 obj.Init(this);
+                obj.OnAnnotationItemClicked += OnAnnotationSelected;
                 _list.Add(obj);
```
Comment (cawke) [inline Assets/MirageXR/Tests/NewUI/ContentListView_v2.cs]:
```suggestion
                var obj = Instantiate(_contentListItemPrefab, _listContent);
                obj.Init(this, OnAnnotationSelected);
                _list.Add(obj);
```

File: Assets/MirageXR/Tests/NewUI/ContentListItem_v2.cs [hunk 2]
```diff
@@ -12,19 +12,25 @@ public class ContentListItem_v2 : MonoBehaviour
...
     private ContentListView_v2 _parentView;
     private ToggleObject _content;
     private ContentType _type;
     private int _from;
     private int _to;
 
+    public delegate void OnAnnotationItemClickedDelegate(ToggleObject annotation);
+
+    public event OnAnnotationItemClickedDelegate OnAnnotationItemClicked;
+
```
Comment (cawke) [inline Assets/MirageXR/Tests/NewUI/ContentListItem_v2.cs]:
```suggestion
    private System.Action<ToggleObject> _onListItemPressed;
```

File: Assets/MirageXR/Tests/NewUI/ContentListItem_v2.cs [hunk 3]
```diff
@@ -12,19 +12,25 @@ public class ContentListItem_v2 : MonoBehaviour
...
     private int _maxStepIndex => activityManager.ActionsOfTypeAction.Count - 1;
 
     public void Init(ContentListView_v2 parentView)
     {
         _parentView = parentView;
         _btnSettings.onClick.AddListener(OnSettingsPressed);
+        _btnListItem.onClick.AddListener(OnListItemPressed);
     }
```
Comment (cawke) [inline Assets/MirageXR/Tests/NewUI/ContentListItem_v2.cs]:
```suggestion
    public void Init(ContentListView_v2 parentView, System.Action<ToggleObject> onListItemPressed)
    {
        _parentView = parentView;
        _onListItemPressed = onListItemPressed;
        _btnSettings.onClick.AddListener(OnSettingsPressed);
        _btnListItem.onClick.AddListener(OnListItemPressed);
    }
```

File: Assets/MirageXR/Tests/NewUI/ContentListItem_v2.cs [hunk 4]
```diff
@@ -53,6 +59,12 @@ private void OnSettingsPressed()
             ("Rename", RenameContent, false),
             ($"Keep alive {_from}-{_to}", ChangeKeepAlive, false),
             ("Delete", DeleteContent, true));
+        
+    }
+
+    private void OnListItemPressed()
+    {
+        OnAnnotationItemClicked?.Invoke(_content);
     }
```
Comment (cawke) [inline Assets/MirageXR/Tests/NewUI/ContentListItem_v2.cs]:
```suggestion
    private void OnListItemPressed()
    {
        _onListItemPressed?.Invoke(_content);
    }
```

**Review by cawke — CHANGES_REQUESTED** [review]

**Review by cawke — APPROVED** [review]
ty

**Pull request merged**